app/lib/llm_clients/openai/azure_client.rb

```diff
@@ -0,0 +1,32 @@
+require "openai"
+
+module LlmClients
+  module Openai
+    class AzureClient < Client
+      private
+
+      def client_provider
+        "openai_azure"
+      end
+
+      def chat_client
+        az_client(@model)
```

Comment:
This isn't really a "client", is it? It's more of a ClientFactory. Right now, one would call `AzureClient.new(...).chat_client` to get a client, right?

What do you think about leaving this work to class methods, and just making the Client class be a client?

E.g.:

```ruby
class << self
  def chat_client
    new(endpoint, ...)
  end

  def embedding_client
    new(...)
  end
end
```

Reply:
Another option would be to just implement another class for this factory-type work, I'm not stuck on class methods. I just think logic to build clients is better outside the client instance itself.

Reply:
Ah, just bad naming. I was using `client` in the `ruby-openai` sense but since we're defining "LLM client" I should really using "connection". These represent the under-the-hood connectivity.

Reply:
It looks like now the "Client" is the factory, and the "connection" is the client. I was just thinking this class called Client wasn't actually a client in any meaningful way, so it should be called something else.

I think this is an issue in general. Clients are hard to create because of all the parameters you either a) need to have, or b) need to derive. I added a ModelLoader class for the API controllers because it was a pain.

Maybe we should embrace the factory pattern and just call these classes `ClientFactory` or `ClientHelper` or something.

Ah, I think what throws me off about this is that AzureClient _is a_ Client, _and_ it _has a_ Client. Does that seem odd to you?

Reply:
From Archyve's point of view `AzureClient` is the LLM client used by the app. Under the hood we're using a gem who also happens to call their "entity" a "client". To deal with the confusion I'm calling it a "connection" now. So the LLM client uses an underlying connection. :-) I think it's OK.

Reply:
nm, AzureClient is instantiating `OpenAI::Client`, not `Openai::Client`. My mistake.